server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -272,14 +272,33 @@ protected Builder newBuilder(String name, Map<String, Object> params) {
         }
     }
 
-    public abstract static class AbstractGeometryFieldType<Parsed, Processed> extends MappedFieldType {
+    public abstract static class AbstractSearcheableGeometryFieldType extends MappedFieldType {
+
+        protected QueryProcessor geometryQueryBuilder;
+
+        protected AbstractSearcheableGeometryFieldType() {
+        }
+
+        protected AbstractSearcheableGeometryFieldType(AbstractSearcheableGeometryFieldType ref) {
+            super(ref);
+        }
+        public void setGeometryQueryBuilder(QueryProcessor geometryQueryBuilder)  {
+            this.geometryQueryBuilder = geometryQueryBuilder;
+        }
+
+        public QueryProcessor geometryQueryBuilder() {
+            return geometryQueryBuilder;
+        }
+    }
+
+    public abstract static class AbstractGeometryFieldType<Parsed, Processed> extends AbstractSearcheableGeometryFieldType {
```

Comment:
In this subclass we can remove `setGeometryQueryBuilder` and `geometryQueryBuilder` methods?